Source/Plugins/Core/com.equella.core/src/com/tle/web/controls/youtube/YoutubeAttachmentSerializer.java

```diff
@@ -45,10 +46,9 @@ public EquellaAttachmentBean serialize(Attachment attachment) {
     ybean.setThumbUrl((String) cattach.getData(YoutubeUtils.PROPERTY_THUMB_URL));
     ybean.setViewUrl((String) cattach.getData(YoutubeUtils.PROPERTY_PLAY_URL));
     ybean.setCustomParameters((String) cattach.getData(YoutubeUtils.PROPERTY_PARAMETERS));
-    final Long date = (Long) cattach.getData(YoutubeUtils.PROPERTY_DATE);
-    if (date != null) {
-      ybean.setUploadedDate(new Date(date));
-    }
+    final Optional<Date> date =
+        YoutubeUtils.parseDateModifiedToDate((String) cattach.getData(YoutubeUtils.PROPERTY_DATE));
+    date.ifPresent(ybean::setUploadedDate);
```

Comment:
In this case, the intermediate `date` variable is rather redundant. I'd keep it nice and tidy and simply have:

```suggestion
    YoutubeUtils.parseDateModifiedToDate((String) cattach.getData(YoutubeUtils.PROPERTY_DATE))
      .ifPresent(ybean::setUploadedDate);
```